src/librustc_typeck/check/method/probe.rs

```diff
@@ -895,20 +895,46 @@ impl<'a, 'gcx, 'tcx> ProbeContext<'a, 'gcx, 'tcx> {
         let trait_substs = self.fresh_item_substs(trait_def_id);
         let trait_ref = ty::TraitRef::new(trait_def_id, trait_substs);
 
-        for item in self.impl_or_trait_item(trait_def_id) {
-            // Check whether `trait_def_id` defines a method with suitable name:
-            if !self.has_applicable_self(&item) {
-                debug!("method has inapplicable self");
-                self.record_static_candidate(TraitSource(trait_def_id));
-                continue;
+        // If a trait alias, elaborate its bounds magically...
+        // TODO? figure out a better way to determine if this is an alias
+
+        let is_trait_alias = {
+            let id = self.tcx.hir().as_local_hir_id(trait_def_id).unwrap();
+            let item = self.tcx.hir().expect_item_by_hir_id(id);
+            match item.node {
+                hir::ItemKind::TraitAlias(..) => true,
+                _ => false,
             }
+        };
 
-            let (xform_self_ty, xform_ret_ty) =
-                self.xform_self_ty(&item, trait_ref.self_ty(), trait_substs);
-            self.push_candidate(Candidate {
-                xform_self_ty, xform_ret_ty, item, import_id,
-                kind: TraitCandidate(trait_ref),
-            }, false);
+        if is_trait_alias {
+            debug!("assemble_extension_candidates_for_trait; no items, assuming alias, elaborating bounds");
+            self.elaborate_bounds(iter::once(trait_ref.to_poly_trait_ref()), |this, new_trait_ref, item| {
```

Comment:
I think this call is right, but I'm no expert on this area, so perhaps someone else can confirm.